src/models.ts

```diff
@@ -259,6 +269,47 @@ export class BasicFilter extends Filter {
   }
 }
 
+export class BasicFilterWithKeys extends BasicFilter {
+  keyValues: (string | number | boolean)[][];
+  target: IFilterKeyColumnsTarget;
+
+  constructor(
+    target: IFilterKeyColumnsTarget,
+    operator: BasicFilterOperators,
+    values: ((string | number | boolean) | (string | number | boolean)[]),
+    keyValues: (string | number | boolean)[][]
+  ) {
+    super(target, operator, values);
+    this.keyValues = keyValues;
+    this.target = target;
+    let numberOfKeys = target.keys ? target.keys.length : 0;
+
+    if (numberOfKeys && !keyValues) {
```

Comment:
I think that we can accept cases when the user didn't specify keys and keyValues because this enables him to use this class also when the column doesn't have key columns.